doc/compression.md

```diff
@@ -67,16 +68,21 @@ cases.
 
 ### Compression Levels and Algorithms
 
-We currently (as of July 2015) support _gzip_ and _deflate_ as algorithms (with
-the possible future addition of snappy). In order to simplify the public API,
-it's intended to abstract the algorithms as _compression levels_ (such as "low",
-"medium", "high") that'd map to concrete algorithms and/or their settings (such
-as "low" mapping to "gzip -3" and "high" mapping to "gzip -9"). However, we
-can't presently (July 2015) implement said compression levels at the client side
-without either a initial negotiation of capabilities or an automatic retry
-mechanism. Therefore, compression levels are only supported at the server side,
-which is aware of the client's capabilities by virtue of the incoming
-Message-Accept-Encoding header.
+The set of supported algorithm is implementation dependent. In order to simplify
+the public API and to operate seamlessly across implementations (both in terms
+of languages but also different version of the same one), we introduce the idea
+of _compression levels_ (such as "low", "medium", "high").
+
+Levels map to concrete algorithms and/or their settings (such as "low" mapping
+to "gzip -3" and "high" mapping to "gzip -9") automatically depending on what a
+peer is known to support. A server is always aware of what its clients support,
+as clients disclose it in their Message-Accept-Encoding header as part of their
+initial call. A client doesn't a priori (presently) know which algorithms a
+server supports. This issue can be addressed with an initial negotiation of
+capabilities or an automatic retry mechanism. These features will be implemented
+in the future. Currently however, compression levels are only supported at the
+server side, which is aware of the client's capabilities by virtue of the
+incoming.
```

Comment:
you are looking at an old commit. Refer to https://github.com/grpc/grpc/pull/5641/commits/9f26a149a5da48364ec62dd607e34104bb851c6d